libs/methodologies/bold/rule-processors/mass-id/src/hauler-identification/hauler-identification.processor.ts

```diff
@@ -0,0 +1,105 @@
+import type { EvaluateResultOutput } from '@carrot-fndn/shared/rule/standard-data-processor';
+
+import { getOrUndefined, isNil } from '@carrot-fndn/shared/helpers';
+import { getEventAttributeValue } from '@carrot-fndn/shared/methodologies/bold/getters';
+import {
+  and,
+  eventHasMetadataAttribute,
+  eventLabelIsAnyOf,
+  eventNameIsAnyOf,
+} from '@carrot-fndn/shared/methodologies/bold/predicates';
+import { ParentDocumentRuleProcessor } from '@carrot-fndn/shared/methodologies/bold/processors';
+import {
+  type Document,
+  type DocumentEvent,
+  DocumentEventName,
+  DocumentEventVehicleType,
+  NewDocumentEventAttributeName,
+} from '@carrot-fndn/shared/methodologies/bold/types';
+import { RuleOutputStatus } from '@carrot-fndn/shared/rule/types';
+import { MethodologyDocumentEventLabel } from '@carrot-fndn/shared/types';
+
+const { ACTOR, PICK_UP } = DocumentEventName;
+const { HAULER } = MethodologyDocumentEventLabel;
+const { VEHICLE_TYPE } = NewDocumentEventAttributeName;
+
+type Subject = {
+  haulerEvent: DocumentEvent | undefined;
+  pickUpEvent: DocumentEvent | undefined;
+};
+
+export const OPTIONAL_HAULER_VEHICLE_TYPES = [
+  DocumentEventVehicleType.SLUDGE_PIPES,
+  DocumentEventVehicleType.CART,
+] as const;
+
+export const RESULT_COMMENTS = {
+  HAULER_EVENT_FOUND: `A "${HAULER}" event was found.`,
```

Comment:
I've updated the comments in Notion to be more specific
```suggestion
  HAULER_EVENT_FOUND: `An "${ACTOR}" with the label "${HAULER}" was found.`,
```